File: src/main/java/com/deneb/client/command/commands/TP.java
package com.deneb.client.command.commands;

import com.deneb.client.command.Command;
import com.deneb.client.utils.ChatUtil;

/**
 * Created by B_312 on 01/03/21
 * Updated by B_312 on 01/15/21
 */
@Command.Info(command = "tp", description = "Teleport you to the place you want.")
public class TP extends Command {

    @Override
    public void onCall(String s, String[] args) {
        if (args.length < 3) {
            ChatUtil.printErrorChatMessage(getSyntax());
            return;
        }
        try {
            int x = Integer.parseInt(args[0]);
            int y = Integer.parseInt(args[1]);
            int z = Integer.parseInt(args[2]);
            mc.player.setPosition(x, y, z);
        } catch (Exception e) {
            ChatUtil.printErrorChatMessage(getSyntax());
        }
    }

    @Override
    public String getSyntax() {
        return "tp <x> <y> <z>";
    }

}

File: src/main/java/club/deneb/client/command/commands/TP.kt
package club.deneb.client.command.commands

import club.deneb.client.command.Command
import club.deneb.client.utils.ChatUtil.printErrorChatMessage

/**
 * Created by B_312 on 01/03/21
 * Updated by B_312 on 01/15/21
 */
@Command.Info(command = "tp", description = "Teleport you to the place you want.")
class TP : Command() {
    override fun onCall(s: String, vararg args: String) {
        if (args.size < 3) {
            printErrorChatMessage(getSyntax())
            return
        }
        try {
            val x = args[0].toInt()
            val y = args[1].toInt()
            val z = args[2].toInt()
            mc.player.setPosition(x.toDouble(), y.toDouble(), z.toDouble())
        } catch (e: Exception) {
            printErrorChatMessage(getSyntax())
        }
    }

    override fun getSyntax(): String {
        return "tp <x> <y> <z>"
    }
}